hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -192,6 +273,63 @@ class IRSuite extends HailSuite {
     assertEvalsTo(ApplyUnaryPrimOp(BitNot(), i64na), null)
   }
 
+  @Test def testApplyUnaryPrimOpInferPType() {
+    val i32na = NA(TInt32())
+    val i64na = NA(TInt64())
+    val f32na = NA(TFloat32())
+    val f64na = NA(TFloat64())
+    val bna = NA(TBoolean())
+
+    var node = ApplyUnaryPrimOp(Negate(), I32(5))
+    assertPType(node, PInt32())
+    node = ApplyUnaryPrimOp(Negate(), i32na)
+    // TODO: this feels like there should be a NULL/missing type, or at least a missing flag on the PType
```

Comment:
can you explain?

Reply:
If it's missing it seems like we can generate different code for it, and in the case of region, avoid allocating, getting, and de-referencing the value, since we know its value explicitly (can be a compile-time value added to bytecode/const). I wrote hoping that you would notice and ask me, so that I could better understand code-generation as it relates to this situation.